### ISSUE TITLE
"Lint reasons are experimental" not flagged by clippy for rust-version without support for them

### ISSUE THREAD

**Issue by IceTDrinker** [rust-lang/rust-clippy#13406]
### Summary

Looked for issues with "rust-version reason" could not find anything.

Asked on zulip here https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/Lint.20reasons.20are.20experimental.20not.20detected.20with.20rust-version

reproduced here

```
Hello I have a project with

rust-version = "1.76"

but it does not seem to detect that allow attributes with "reason" is not available in 1.76, it does not raise an issue in a clippy run, does that sound like a bug ?
```

was advised to open an issue if I did not find one in the clippy issues. The person advising to open an issue cites "language properties" (like this case) and "trait implementations" as being things clippy does not track very well

Cheers

### Reproducer

I tried this code:

```rust
#[allow(dead_code, reason = "reproducer for the issue")]
struct MyThing {
    unused_field: bool
}
```

I expected to see this happen:

Get a warning from clippy given the min rust-version of the project is 1.76 and does not support "allow with reason" (stabilized in 1.81)

Instead, this happened:

No warnings

### Version

```text
rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-unknown-linux-gnu
release: 1.81.0
LLVM version: 18.1.7
```

**Comment by Jarcho** [comment]
We don't have any lint that would cover this right now. `incompatible_msrv` only covers library items, not language features.
